### DISCUSSION TITLE
0.1.3

### DISCUSSION THREAD

**Discussion by samad-yar-khan** [middlewarehq/middleware#440] [Announcements]
### Enhancements and Features 🚀
- **GraphQL API for User and Org Repositories**: Introduced a new GraphQL API to fetch user and organization repositories, enhancing data retrieval capabilities (#360).
- **New Docker Instructions**: Added instructions in the README for creating Docker volumes to improve setup guidance (#355).
- **Deployment Analytics Refactor**: Refactored `deployment_analytics.ts` API endpoint and `fetchTeamDeployments` thunk to streamline code and improve functionality (#410).
- **App Version Management**: Added mechanisms to save app versions and check for new releases, ensuring users stay updated with the latest features and fixes (#401).
- **CLI Version Checker**: Introduced a CLI version checker to verify the current version and updates efficiently (#420).
- **Automated Code Generation**: Automated the code generation process for new configuration settings, reducing manual coding efforts (#433).
- **Date Handling in UI**: Refactored date handling in `WeeklyDeliveryVolumeCard` for better readability and accuracy (#412).
- **Configuration of Sync Days**: Made default sync days configurable via environment variables, adding flexibility to synchronization settings (#400).

### Fixes and Improvements 🛠️
- **Image Loader and Pagination Fixes**: Fixed the image loader for PAT-token and improved pagination for fetching organization repositories (#356).
- **Sync Loader Enhancements**: Improved sync loader animation and created a loading state card to provide better user feedback during data synchronization (#386).
- **Sorting and Spacing Adjustments**: Addressed sorting issues in the PR table and fixed spacing and alignment in the `TeamsList` component (#388, #363).
- **Loading State Updates**: Enhanced loading states for various API searches and synchronization processes, providing a smoother user experience (#373, #384, #417).
- **Repository Bookmarking**: Ensured bookmarks are saved irrespective of repository data availability (#380).
- **Deployment Frequency Calculation**: Fixed the calculation of deployment frequency for more accurate metrics (#382).
- **Link and API Endpoint Updates**: Updated links for DORA metrics and fixed broken links to external MTTR resources, ensuring all references are current and accurate (#362, #391).

### Infrastructure and Maintenance 🔧
- **Docker Improvements**: Implemented Docker Compose build before watch to sync local file changes, and mapped ports only to localhost to enhance security and efficiency (#435, #437).
- **Log Management**: Separated logs for the sync server to streamline log analysis and troubleshooting (#368).
- **Code Refactoring and Tests**: Refactored various components, including jest configuration and utility filters, to improve code maintainability and performance (#427, #409).
- **Revert and Adjust Access Limits**: Reverted the limitation of access for database and backend servers in the developer environment and adjusted port mappings and environment variable access (#436, #438).

### Documentation 📃
- **Readme Updates**: Updated README with new links to DORA metrics, Docker instructions, and general improvements to keep documentation current and useful (#365, #394).

### Dependency Updates 📦
- **Dependency Management**: Updated multiple dependencies to ensure the latest security patches and feature enhancements are applied (#419).

### New Contributors ⭐ 
* @gajrajgchouhan made their first contribution in #367
* @nathenharvey made their first contribution in #394
* @ayush3160 made their first contribution in #411
* @VipinDevelops made their first contribution in #429

###  All Contributors 🤝 
- @aravindputrevu
- @adnanhashmi09
- @e-for-eshaan
- @amoghjalan
- @shivam-bit
- @samad-yar-khan
- @gajrajgchouhan
- @nathenharvey
- @ayush3160
- @VipinDevelops

**Full Changelog**: https://github.com/middlewarehq/middleware/compare/0.1.2...0.1.3

<hr /><em>This discussion was created from the release <a href='https://github.com/middlewarehq/middleware/releases/tag/0.1.3'>0.1.3</a>.</em>

**Comment by aravindputrevu** [comment]
Thank you Samad and team
